offer-manager-domain/src/main/groovy/life/qbic/business/customers/create/CreateCustomerDataSource.groovy

```diff
@@ -21,4 +22,38 @@ interface CreateCustomerDataSource {
      * @since 1.0.0
      */
     void addCustomer(Customer customer) throws DatabaseQueryException
+
+    /**
+     * Updates the information of a given customer specified by a customer ID
+     *
+     * @param customerId to specify the customer to be updated
+     * @param updatedCustomer customer containing all information and the updates of a customer
+     * @throws DatabaseQueryException When a customer could not been updated in the customer
+     * database
+     * @since 1.0.0
+     */
+    void updateCustomer(String customerId, Customer updatedCustomer) throws DatabaseQueryException
+
+    /**
+     * Returns a customer given a customer specified by a customer ID
+     *
+     * @param customerId to specify and existing customer
+     */
+    Customer getCustomer(int customerId)
+
+    /**
+     * Searches for a customer in a database and returns its id
+     *
+     * @param customer The customer that needs to be searched in the database
+     * @return the customer id or -1 if not found
+     */
+    Optional<Integer> findCustomer(Customer customer)
```

Comment:
Why is this Optional if an integer (-1) is returned in any case?

Reply:
Its an outdated description, good catch I will update it :+1: